docs/modules/ROOT/pages/explanations/iamkey.adoc

```diff
@@ -0,0 +1,19 @@
+== IAMKeys permissions
+
+The IAMkeys have restricted access to exoscale API.
+The Object Storage `sos` service is the only service the keys have access to.
+The following are the allowed operations on buckets:
+
+- abort-sos-multipart-upload
+- delete-sos-bucket
```

Comment:
We explictly removed the permission to create buckets. Shouldn't we also remove the `delete-sos-bucket`? The bucket is being deleted with the secret as defined in the provider config, not the credentials of the IAM key itself.

Reply:
That's a good point you raised. In normal conditions the client would not know the names of buckets it does not have access to which may be a valid point to leave the deletion in there but we also put our provider in difficult situation when the bucket suddenly disappears while it tries to reconcile the CRD.  @Kidswiss what's your take on that?

Reply:
@ZuGao Do you know how the provider would react if the bucket is deleted? Would it try to re-create it, or simply error out?

If it's the latter, I'd remove the permission to delete, as it could lead to deadlocks for a customer.

Reply:
Actually the bucket is being recreated without any issues :)

Reply:
I don't follow what you discuss now. This is about the IAM key having more permissions than it needs to. Bucket deletion and creation happens with a different credentials.

Reply:
It does not matter which credentials the provider is using. We are trying to figure out what happens if the user would delete its own buckets outside the provider control with the IAM key provided to him, for instance with an s3 client. How would the provider react? Well I figured out that the provider would recreate it with no issues, so we could actually leave the delete in place but nevertheless I believe we should remove this operation as you suggested. We should leave the deletion of buckets only via the CRD manifest or via its claim.

Reply:
Fine by me.

Reply:
I updated the operation list.

Reply:
Thanks for the explanation. In that case I would opt to allow creation and deletion only via claim to have exactly 1 supported way of creating and deleting buckets.

Reply:
> I updated the operation list.

Not in the docs :)

Reply:
yeah... fixed it. Do I leave this PR open until Exoscale fixes the issue?

Reply:
> until Exoscale fixes the issue?

What issue?

Reply:
I wrote in the description about a bug in exoscale.

Reply:
ah, I missed/forgot that one :)

For me it's fine to merge, we probably don't know how long it takes for the issue to be fixed. I wouldn't keep this open for weeks and the remaining operations can be re-added later as well.

Reply:
I will open an issue in GitHub then so we have a ticket in our system. Please approve it so I can merge it

Reply:
Opened [issue]( https://github.com/vshn/provider-exoscale/issues/19)